Makefile: separate GOFLAGS and GOMODVENDORFLAGS

otan opened the pull request:
GOFLAGS is overwritten by various scripts in the release process. This
causes the [Publish Bleeding Edge](https://teamcity.cockroachdb.com//viewLog.html?buildId=2035695&buildTypeId=Cockroach_MergeToMaster)
to fail.

This change modifies the Makefile to introduce a secondary flag for
-mod=vendor that ensures it is always used.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50615)
<!-- Reviewable:end -->

otan commented:
Publish bleeding edge watch: https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_MergeToMaster&branch_Cockroach=50615&tab=buildTypeStatusDiv

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50615#-:-MAbMPEpCY8UPE-ZvMT8:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @petermattis)



<!-- Sent from Reviewable.io -->

otan committed:
Makefile: separate GOFLAGS and GOMODVENDORFLAGS

GOFLAGS is overwritten by various scripts in the release process. This
causes the [Publish Bleeding Edge](https://teamcity.cockroachdb.com//viewLog.html?buildId=2035695&buildTypeId=Cockroach_MergeToMaster)
to fail.

This change modifies the Makefile to introduce a secondary flag for
-mod=vendor that ensures it is always used.

Release note: None

otan commented:
bors r=petermattis
